src/main/java/co/cask/http/HttpMethodInfo.java

```diff
@@ -114,7 +115,8 @@ private void bodyConsumerChunk(ChannelBuffer buffer) throws HandlerException {
     try {
       bodyConsumer.chunk(buffer, responder);
     } catch (Throwable t) {
-      throw bodyConsumerError(t);
+      bodyConsumerError(t);
```

Comment:
yeah i guess we should do that, though its a bit weird on which exception the exception handler should get in that case. I guess it should get the handleError exception